Support Undo/Redo API

ndoschek opened the pull request:
- Based on PR https://github.com/eclipse-emfcloud/emfcloud-modelserver/pull/49

Signed-off-by: Nina Doschek <ndoschek@eclipsesource.com>

martin-fleck-at reviewed (changes requested):
Change looks good to me for the most part, thank you very much Nina! I just have a request: Could you also maybe add an 'edit', 'undo', and 'redo' command for any other file than the SuperBrewer3000.coffee? That way we could test that the editing domains are truly separate since you can undo/redo changes on two different files. What do you think?

ndoschek commented:
Thanks for your review so far Martin, and that's a fair point!
I added separate test menus for each resource to be able to test different scenarios regarding the division in separate ResourceSets. For more general requests to the `ModelServer` i kept the existing menu, which holds the calls `ping`, `getModelUris` and so on.
In the course of the change, I noticed that the notifications on subsription updates were not distinct enough (i.e. newer notification with the same message perhaps, were not shown on top of the notification panel).
So I took a deeper look at the whole framework and found some other inconsistencies and cleared things up a bit with this change.
I introduced the `SubscriptionListener` to be able to register multiple listeners if necessary. Also I did some renaming and code clean up. Altogether it was my goal to make the project a little bit more clearer and self-explanatory and increase testability.

Regarding the notifications, now they show the current timestamp and the response. If it is a subscription update, the linked modelUri  is shown as well. It is now easier to track updates and responses from the ModelServer.
(Hint: clicking the :bell: symbol in theia in the lower right corner keeps the notification panel up and the notifications won't disappear after the default timeout)

martin-fleck-at reviewed (approved):
Change looks very good to me and works well with the corresponding server change, thank you Nina! I am a huge fan of the new split menus as we already offer quite a large range of commands for each file, so very good work and thank you very much!

I had some minor comment on the server-side change so I will wait until that is fixed as well before merging this.
